tools/azsdk-cli/auto-documentation.yml

```diff
@@ -0,0 +1,40 @@
+jobs:
+  - job: UpdateMCPDocumentation
+    displayName: 'Update MCP Tools Documentation'
+    pool:
+      name: $(LINUXPOOL)
+      image: $(LINUXVMIMAGE)
+      os: linux
+    
+    steps:
+    - checkout: self
+      displayName: 'Checkout current branch'
+    
+    - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+      parameters:
+        Repositories:
+          - Name: Azure/azure-sdk-tools
+            Commitish: main
+            WorkingDirectory: $(System.DefaultWorkingDirectory)/azure-sdk-tools-main
+        Paths:
+          - 'tools/azsdk-cli/**'
```

Comment:
why do we need this one when repo is already checked out at line 10? All of the below steps are using default checked out path and not this sparse checked path.